kernel/src/platform/platform.rs

```diff
@@ -131,9 +131,137 @@ pub trait SyscallFilter {
     }
 }
 
-/// Implement default SyscallFilter trait for unit.
+/// Implement default allow all SyscallFilter trait for unit.
 impl SyscallFilter for () {}
 
+/// Implement default SyscallFilter trait for filtering based on the TBF header.
+pub struct TbfHeaderFilterDefaultAllow {}
+
+/// Implement default SyscallFilter trait for filtering based on the TBF header.
+///
+/// This will check if the process has TbfHeaderPermissions specified.
+/// If the process has TbfHeaderPermissions they will be used to determine
+/// access permissions. For details on this see the TockBinaryFormat
+/// documentation.
+/// If no permissions are specified the default is to allow the syscall.
```

Comment:
Do these comments show up anywhere? In any case, we generally put the comments on the actual type.